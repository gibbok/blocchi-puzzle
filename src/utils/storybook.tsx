import React from 'react';
import GlobalStyle from '../components/GlobalStyle';
import styled from 'styled-components';
import { wood } from '../assets/images';

/**
 * Decorate a story with glocal styles.
 * @param story Story
 */
export const globalStylesDecorator = (story: () => unknown): JSX.Element => (
  <React.Fragment>
    <GlobalStyle />
    {story()}
  </React.Fragment>
);

const Bkg = styled.div`
  background-image: url(${wood});
  width: 100vw;
  height: 100vh;
`;

/**
 * Decorate a story with a page background.
 * @param story Story
 */
export const pageBackground = (story: () => React.ReactNode): JSX.Element => <Bkg>{story()}</Bkg>;
