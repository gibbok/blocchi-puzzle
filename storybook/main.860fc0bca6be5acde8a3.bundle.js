(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{498:function(n,o,p){p(499),p(646),n.exports=p(1404)},562:function(n,o){}},[[498,1,2]]]);