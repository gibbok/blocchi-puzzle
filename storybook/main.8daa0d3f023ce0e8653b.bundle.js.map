{"version":3,"file":"main.8daa0d3f023ce0e8653b.bundle.js","sources":["webpack:///main.8daa0d3f023ce0e8653b.bundle.js"],"mappings":"AAAA;;AAwEA;;;;;;;AAOA;;;AAGA;;;AAGA;;;AAGA;AACA;;;AAGA;;;;AAKA;;;;AAQA;;;;;;;;;;;;;;;;;;;;;;AAsBA;;;;AAKA;;;;AAwHA;AACA;;;;AAIA;;AAEA;;;;AAIA;;;AAGA;;;;;AAMA;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAwCA;;AA8CA;;;;;;;AAOA;AAGA;AAKA;;;;;AAKA;AACA;;AAPA;AAWA;;;;AAIA;AACA;;AAhBA;AAoBA;;;;AAIA;;AAEA;;AA1BA;AA8BA;;;;;AAKA;;AAEA;;;;;;;AArCA;AA8CA;;AAEA;AACA;AACA;AACA;AAnDA;;AAuDA;;;AAGA;AAWA;;;;;AA+CA;;;;;;;;;;;;;AAaA;;;;AAIA;;;;AAKA;;;;;;AAWA;;AA+YA;;;;AAyDA;AAAA;;AASA;;;;AA6BA;;;;;AA2CA;;;AAGA;;;AAaA;;;;AAuBA;;AAEA;;;;AAIA;;;;AAIA;;;;;;;;;AAiBA;;;;;;;;;;AA+BA;;;;;;;;;AASA;;AAGA;AACA;;;;;;;AAQA;;AAMA;AAmJA;;;AAQA;AAm9BA;;;;;;;;;;;;;;;;;;;;;;;;;AA0BA;AA0GA;AACA;;;;;;;AAQA;;;;;;;;;;AA0DA;;;;;;;AAQA;;;AAGA;;AAEA;;;AAGA;;;AAIA;;AAGA;;;;;AAqEA;;;;;;;AAkOA;;AAEA;;;AAGA;;;AAIA;;;;;;;;;;;AAuCA;;AAGA;;AAQA;;AAmLA;;;AAmIA;;;AA4DA;;;AAIA","sourceRoot":""}