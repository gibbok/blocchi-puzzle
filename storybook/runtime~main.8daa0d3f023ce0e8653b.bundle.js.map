{"version":3,"file":"runtime~main.8daa0d3f023ce0e8653b.bundle.js","sources":["webpack:///runtime~main.8daa0d3f023ce0e8653b.bundle.js"],"mappings":"AAAA","sourceRoot":""}