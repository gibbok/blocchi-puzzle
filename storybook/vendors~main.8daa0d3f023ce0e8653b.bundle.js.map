{"version":3,"file":"vendors~main.8daa0d3f023ce0e8653b.bundle.js","sources":["webpack:///vendors~main.8daa0d3f023ce0e8653b.bundle.js"],"mappings":";AAAA;;AAySA;;AAOA","sourceRoot":""}