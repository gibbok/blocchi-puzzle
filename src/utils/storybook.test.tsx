import React from 'react';
import { globalStylesDecorator, pageBackground } from './storybook';
import renderer from 'react-test-renderer';

describe('storybook', () => {
  describe('globalStylesDecorator', () => {
    it('should render story with global styles in document head', () => {
      const tree = renderer.create(globalStylesDecorator(() => <div>test</div>));
      expect(document.head).toMatchSnapshot();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('pageBackground', () => {
    it('should render story within a wrapper element with background image', () => {
      const tree = renderer.create(pageBackground(() => <div>test</div>));
      expect(tree).toMatchSnapshot();
    });
  });
});
