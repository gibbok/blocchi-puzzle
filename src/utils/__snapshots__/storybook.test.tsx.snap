// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storybook globalStylesDecorator should render story with global styles in document head 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.1.1"
  >
    html{line-height:1.15;-webkit-text-size-adjust:100%;}
    body{margin:0;}
    main{display:block;}
    h1{font-size:2em;margin:0.67em 0;}
    hr{box-sizing:content-box;height:0;overflow:visible;}
    pre{font-family:monospace,monospace;font-size:1em;}
    a{background-color:transparent;}
    abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}
    b,strong{font-weight:bolder;}
    code,kbd,samp{font-family:monospace,monospace;font-size:1em;}
    small{font-size:80%;}
    sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}
    sub{bottom:-0.25em;}
    sup{top:-0.5em;}
    img{border-style:none;}
    button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}
    button,input{overflow:visible;}
    button,select{text-transform:none;}
    button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}
    button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}
    button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}
    fieldset{padding:0.35em 0.75em 0.625em;}
    legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}
    progress{vertical-align:baseline;}
    textarea{overflow:auto;}
    [type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}
    [type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}
    [type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}
    [type="search"]::-webkit-search-decoration{-webkit-appearance:none;}
    ::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}
    details{display:block;}
    summary{display:list-item;}
    template{display:none;}
    [hidden]{display:none;}
    html{box-sizing:border-box;font-size:62.5%;font-family:'Oleo Script',cursive;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%;-o-font-feature-settings:'salt' 1;-ms-font-feature-settings:'salt' 1;-webkit-font-smoothing:antialiased;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;-moz-osx-font-smoothing:grayscale;-moz-font-feature-settings:'salt' 1;-webkit-font-feature-settings:'salt' 1;font-feature-settings:'salt' 1;text-rendering:optimizeLegibility;}
    *,*:before,*:after{box-sizing:inherit;}
    body{margin:0;padding:0;}
  </style>
</head>
`;

exports[`storybook globalStylesDecorator should render story with global styles in document head 2`] = `
<div>
  test
</div>
`;

exports[`storybook pageBackground should render story within a wrapper element with background image 1`] = `
<div
  className="sc-AxjAm fAxUdK"
>
  <div>
    test
  </div>
</div>
`;
